.github/workflows/release.yml

```diff
@@ -27,40 +68,76 @@ jobs:
+          # It allows any application to read the keychain, which is fine in an ephemeral environment,
+          # but not fine if you run this on your own machine.
+          security import certificate.p12 -k build.keychain -P $MACOS_CERTIFICATE_PWD -A -T /usr/bin/codesign -T /usr/bin/productbuild -T /usr/bin/security
+          security set-key-partition-list -S apple-tool:,apple:,codesign: -s -k $TEMP_PWD build.keychain
 
-      - name: "Generate SHA512 for Linux"
-        if: runner.os == 'Linux'
-        run: sha512sum ${{ steps.fn.outputs.FILENAME }} > ${{ steps.fn.outputs.FILENAME }}.sha512
-      - name: "Generate SHA512 for Windows"
-        if: runner.os == 'Windows'
-        shell: bash
-        run: certutil -hashfile ${{ steps.fn.outputs.FILENAME }} SHA512 >> ${{ steps.fn.outputs.FILENAME }}.sha512
-      - name: "Generate SHA512 for OSX"
-        if: runner.os == 'macOS'
-        run: openssl dgst -sha512 > ${{ steps.fn.outputs.FILENAME }}.sha512
+      - name: "Create executable with jpackage (and sign, on MacOS)"
+        uses: ./.github/actions/gradle-and-sha
+        with:
+          gradle-command: jpackage
+          intermediate-filepath: ${{ steps.jpackagefn.outputs.FILEPATH }}
+          final-filepath: ${{ steps.exefn.outputs.FILEPATH }}
+
+      - name: "Notarize app bundle"
+        if: matrix.os == 'macOS-latest'
+        env:
+          MACOS_NOTARIZATION_APPLE_ID: ${{ secrets.MACOS_NOTARIZATION_APPLE_ID }}
+          MACOS_NOTARIZATION_TEAM_ID: ${{ secrets.MACOS_NOTARIZATION_TEAM_ID }}
+          MACOS_NOTARIZATION_PWD: ${{ secrets.MACOS_NOTARIZATION_PWD }}
+          IDENTITY_PUBLIC_KEY: A257HB4NS4
+        run: |
+          echo "Unlock keychain"
+          security unlock-keychain -p temporary-password-to-avoid-GUI-prompt build.keychain
+          security set-key-partition-list -S apple-tool:,apple:,codesign: -s -k temporary-password-to-avoid-GUI-prompt build.keychain
+          echo "Create keychain profile"
+          xcrun notarytool store-credentials "notarytool-profile" --apple-id "$MACOS_NOTARIZATION_APPLE_ID" --team-id "$MACOS_NOTARIZATION_TEAM_ID" --password "$MACOS_NOTARIZATION_PWD"
+          echo "Creating temp notarization archive"
+          ditto -c -k --sequesterRsrc --keepParent ${{ steps.exefn.outputs.FILEPATH }} "notarization.zip"
+          echo "Notarize app -- this may take a few minutes"
+          xcrun notarytool submit "notarization.zip" --keychain-profile "notarytool-profile" --wait
+          echo "Attach staple"
+          xcrun stapler staple ${{ steps.exefn.outputs.FILEPATH }}
 
       - uses: actions/upload-artifact@v3
         with:
           name: Package
           if-no-files-found: error
           path: |
-            ${{ github.workspace }}/${{ steps.fn.outputs.FILENAME }}
-            ${{ github.workspace }}/${{ steps.fn.outputs.FILENAME }}.sha512
+            ${{ github.workspace }}/${{ steps.zipfn.outputs.FILEPATH }}
```

Comment:
I think I asked about this previously but forgot the answer... I see in the test run [here](https://github.com/BrightSpots/rcv/actions/runs/5213169580) that it creates a single file called "Package", with all three versions in it. Is it possible to upload the individual versions and their hashes separately instead of all bundled together?

Reply:
Only in a Release, but not in an Artifact. So here, it would be possible: https://github.com/BrightSpots/rcv/releases/tag/

And it is indeed what happens there -- just not in the Artifact.